TebLocalPlannerROS:

  odom_topic: /load/odom  # This parameters changes with respect to load. The broadcaster is incharge to make the right transformation
  map_frame:   /map #/load/base_link #
      
  # Trajectory
    
  teb_autosize: False
  dt_ref: 0.1
  dt_hysteresis: 0.0
  global_plan_overwrite_orientation: False
  allow_init_with_backwards_motion: False
  max_global_plan_lookahead_dist: 0.5 # This parameters changes with respect to load
  feasibility_check_no_poses: 5
  global_plan_viapoint_sep: 0.2
      
  # Robot
          
  max_vel_x: 1.4   # This parameters changes with respect to load
  max_vel_x_backwards: 0.2
  max_vel_y: 0.0
  max_vel_theta: 0.5
  acc_lim_x: 0.7
  acc_lim_theta: 0.2
  min_turning_radius: 0.0 # diff-drive robot (can turn on place!)

  footprint_model: # types: "point", "circular", "line", "two_circles", "polygon"
    type: "polygon"
    radius: 0.2 # for type "circular"
    line_start: [0.0, 0.0] # for type "line". Modify also min_obstacle_dist!
    line_end: [0.9, 0.0] # for type "line". Modify also min_obstacle_dist!
    front_offset: 0.2 # for type "two_circles"
    front_radius: 0.2 # for type "two_circles"
    rear_offset: 0.2 # for type "two_circles"
    rear_radius: 0.2 # for type "two_circles"
    vertices: [[-0.36, 0.36], [0.36, 0.36], [0.36, -0.36], [-0.36, -0.36]] # for type "polygon" # This parameters changes with respect to load
    
     
  # GoalTolerance
      
  xy_goal_tolerance: 0.09 # This parameters changes with respect to load
  yaw_goal_tolerance: 0.04 # This parameters changes with respect to load
  free_goal_vel: False
      
  # Obstacles
      
  min_obstacle_dist: 0.01   # 0.01 for plygon type. 0.5 for line type
  include_costmap_obstacles: True
  costmap_obstacles_behind_robot_dist: 1.0
  obstacle_poses_affected: 30
  obstacle_association_force_inclusion_factor: 1.5
  costmap_converter_plugin: ""
  costmap_converter_spin_thread: True
  costmap_converter_rate: 5

  # Optimization
      
  no_inner_iterations: 5
  no_outer_iterations: 4
  optimization_activate: True
  optimization_verbose: False
  penalty_epsilon: 0.1
  weight_max_vel_x: 2
  weight_max_vel_theta: 1
  weight_acc_lim_x: 1
  weight_acc_lim_theta: 1
  weight_kinematics_nh: 1000
  weight_kinematics_forward_drive: 1
  weight_kinematics_turning_radius: 1
  weight_optimaltime: 1
  weight_obstacle: 50
  weight_viapoint: 10
  weight_dynamic_obstacle: 10 # not in use yet
  weight_adapt_factor: 2
  

  

  # Homotopy Class Planner

  enable_homotopy_class_planning: False
  enable_multithreading: True
  simple_exploration: False
  max_number_classes: 2
  selection_cost_hysteresis: 1.0
  selection_obst_cost_scale: 1.0
  selection_alternative_time_cost: False
  
  roadmap_graph_no_samples: 10
  roadmap_graph_area_width: 5
  h_signature_prescaler: 0.5
  h_signature_threshold: 0.1
  obstacle_keypoint_offset: 0.1
  obstacle_heading_threshold: 0.45
  visualize_hc_graph: False

  
# Move base parameters
recovery_behavior_enabled: false
clearing_rotation_allowed: false
max_planning_retries: 0
controller_frequency: 10.0
global_frame: /map  

















#TebLocalPlannerROS:

  #odom_topic: /ropod/odom
  #map_frame:   /map #/load/base_link #
      
  ## Trajectory
    
  #teb_autosize: False
  #dt_ref: 0.1
  #dt_hysteresis: 0.0
  #global_plan_overwrite_orientation: False
  #allow_init_with_backwards_motion: False
  #max_global_plan_lookahead_dist: 1.0
  #feasibility_check_no_poses: 10
  #global_plan_viapoint_sep: 0.1
      
  ## Robot
          
  #max_vel_x: 1.2
  #max_vel_x_backwards: 0.2
  #max_vel_y: 0.0
  #max_vel_theta: 0.7
  #acc_lim_x: 0.5
  #acc_lim_theta: 0.5
  #min_turning_radius: 0.0 # diff-drive robot (can turn on place!)

  #footprint_model: # types: "point", "circular", "line", "two_circles", "polygon"
    #type: "polygon"
    #radius: 0.2 # for type "circular"
    #line_start: [0.0, 0.0] # for type "line". Modify also min_obstacle_dist!
    #line_end: [0.9, 0.0] # for type "line". Modify also min_obstacle_dist!
    #front_offset: 0.2 # for type "two_circles"
    #front_radius: 0.2 # for type "two_circles"
    #rear_offset: 0.2 # for type "two_circles"
    #rear_radius: 0.2 # for type "two_circles"
    #vertices: [[-0.33, 0.33], [0.33, 0.33], [0.33, -0.33], [-0.33, -0.33]] # for ropod alone # type "polygon"
    
     
  ## GoalTolerance
      
  #xy_goal_tolerance: 0.2
  #yaw_goal_tolerance: 0.05
  #free_goal_vel: False
      
  ## Obstacles
      
  #min_obstacle_dist: 0.01   # 0.01 for plygon type. 0.5 for line type
  #include_costmap_obstacles: True
  #costmap_obstacles_behind_robot_dist: 1.0
  #obstacle_poses_affected: 30
  #inflation_dist: 0.0
  #obstacle_association_force_inclusion_factor: 1.5
  #costmap_converter_plugin: ""
  #costmap_converter_spin_thread: True
  #costmap_converter_rate: 5

  ## Optimization
      
  #no_inner_iterations: 5
  #no_outer_iterations: 4
  #optimization_activate: True
  #optimization_verbose: False
  #penalty_epsilon: 0.1
  #weight_max_vel_x: 2
  #weight_max_vel_theta: 1
  #weight_acc_lim_x: 1
  #weight_acc_lim_theta: 1
  #weight_kinematics_nh: 1000
  #weight_kinematics_forward_drive: 1
  #weight_kinematics_turning_radius: 1
  #weight_optimaltime: 1
  #weight_obstacle: 10
  #weight_viapoint: 1.0
  #weight_dynamic_obstacle: 10 # not in use yet
  #weight_adapt_factor: 2
  

  

  ## Homotopy Class Planner

  #enable_homotopy_class_planning: False
  #enable_multithreading: True
  #simple_exploration: False
  #max_number_classes: 2
  #selection_cost_hysteresis: 1.0
  #selection_obst_cost_scale: 1.0
  #selection_alternative_time_cost: False
  
  #roadmap_graph_no_samples: 10
  #roadmap_graph_area_width: 5
  #h_signature_prescaler: 0.5
  #h_signature_threshold: 0.1
  #obstacle_keypoint_offset: 0.1
  #obstacle_heading_threshold: 0.45
  #visualize_hc_graph: False

  
## Move base parameters
#recovery_behavior_enabled: false
#clearing_rotation_allowed: false
#max_planning_retries: 0
#controller_frequency: 10.0
#global_frame: /map  